libs/components/src/lib/toggletip/toggletip.spec.ts

```diff
@@ -0,0 +1,162 @@
+			expect(toggletipDefinition()).toBeInstanceOf(FoundationElementRegistry);
+			expect(element).toBeInstanceOf(Toggletip);
+
+			expect(element.open).toBe(false);
+			expect(element.alternate).toBe(false);
+			expect(element.headline).toBeUndefined();
+			expect(element.placement).toBe('right');
+		});
+	});
+
+	describe('open/close', () => {
+		it('should open when its anchor is clicked', async () => {
+			element.anchor = 'anchorButton';
+			await elementUpdated(element);
+
+			expect(element.open).toBe(false);
+
+			anchor.dispatchEvent(new MouseEvent('click', {bubbles: true}));
+			await elementUpdated(element);
+
+			expect(element.open).toEqual(true);
+		});
+
+		it('should remain open when clicked inside', async () => {
+			element.anchor = 'anchorButton';
+			element.open = true;
+			await elementUpdated(element);
+
+			element.dispatchEvent(new MouseEvent('click', {bubbles: true}));
+			await elementUpdated(element);
+
+			expect(element.open).toEqual(true);
+		});
+
+		it('should close when clicked outside', async () => {
+			element.anchor = 'anchorButton';
+			element.open = true;
+			await elementUpdated(element);
+
+			document.body.dispatchEvent(new MouseEvent('click', {bubbles: true}));
+			await elementUpdated(element);
+
+			expect(element.open).toEqual(false);
+		});
+
+		it('should close when Escape is pressed', async () => {
+			element.anchor = 'anchorButton';
+			element.open = true;
+			await elementUpdated(element);
+
+			document.body.dispatchEvent(new KeyboardEvent('keydown', {key: 'Escape', bubbles: true}));
+			await elementUpdated(element);
+
+			expect(element.open).toEqual(false);
+		});
+	
+	});
+
+	describe('anchor', () => {
+		it('should accept an HTMLElement as anchor', async () => {
+			element.anchor = anchor;
+			await elementUpdated(element);
+
+			anchor.dispatchEvent(new MouseEvent('click', {bubbles: true}));
+			await elementUpdated(element);
+
+			expect(element.open).toEqual(true);
+		});
+
+		it('should open correctly when its anchor is changed', async () => {
+			const anchor2 = fixture(
+				'<vwc-button id="anchorButton2"></vwc-button>', ADD_TEMPLATE_TO_FIXTURE
+			) as Button;
+	
+			element.anchor = 'anchorButton';
+			await elementUpdated(element);
+
+			element.anchor = 'anchorButton2';
+			await elementUpdated(element);
+			expect(element.open).toEqual(false);
+
+			anchor.dispatchEvent(new MouseEvent('click', {bubbles: true}));
+			await elementUpdated(element);
+			expect(element.open).toEqual(false);
+
+			anchor2.dispatchEvent(new MouseEvent('click', {bubbles: true}));
+			await elementUpdated(element);
+			expect(element.open).toEqual(true);
+		});
+
+		it('should update the anchor aria-label when none is set', async () => {
+			element.anchor = anchor;
+			await elementUpdated(element);
+
+			expect(anchor.ariaLabel).toEqual(' ; Show more information');
+
+			element.anchor = '';
+			await elementUpdated(element);
+
+			expect(anchor.ariaLabel).toEqual('');
+		});
```

Comment:
Standard is expectations in the end. Usually, a test like this means there are two tests in one. Like this:

```suggestion
		it('should remove the aria-label from old anchor when changed', async () => {
			element.anchor = anchor;
			await elementUpdated(element);

			expect(anchor.ariaLabel).toEqual(' ; Show more information');

			element.anchor = 'anchor2';
			await elementUpdated(element);

			expect(anchor.ariaLabel).toEqual('');
		});
		
		it('should set the aria-label on anchor', async () => {
			element.anchor = anchor;
			await elementUpdated(element);

			expect(anchor.ariaLabel).toEqual(' ; Show more information');
		});
```

Reply:
having multiple `expect`s in a test is reasonable as long as multiple is understood as "small number" and the test is not long.
the suggestion is longer and less readable.

Reply:
"for me"